### PR TITLE
Fix for Granite model, add result visualization, misc updates

### PR DESCRIPTION
**Pull request by HunterGerlach**
- Add perf visualization
- fix plugin for Granite support
- Improve readme guidance
- Add venv edge case support for gitignore
- misc formatting updates

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,13 +38,13 @@ optional arguments:
 
 The tool's behavior can be customized using a YAML configuration file. Take a look at `config.yaml` for an example. More documentation on this should be added in the future.
 
-
 **Results**:
-The tool will produce a results summary logged to stdout, and detailed test results along with its summary in json format.
+The tool will produce a results summary logged to stdout, and detailed test results along with its summary in json format in `outpu/output.json`.
```
Comment (dagrayvid) [inline README.md:42]:
```suggestion
The tool will produce a results summary logged to stdout, and detailed test results along with its summary in json format in `output/output.json`, or the output file specified in the config YAML.
```

File: plugins/openai_plugin.py [hunk 2]
```diff
@@ -100,7 +104,8 @@ def request_http(self, query: dict, user_id: int, test_end_time: float = 0):
             error = message.get("error")
             if error is None:
                 if "/v1/chat/completions" in self.host:
-                    result.output_text = message["choices"][0]['delta']['content']
+                    #result.output_text = message["choices"][0]['delta']['content']
+                    result.output_text = message["choices"][0]['message']['content']
```
Comment (dagrayvid) [inline plugins/openai_plugin.py:108]: @ccamacho, I haven't used the chat completions API, but from https://platform.openai.com/docs/api-reference/chat/create, it seems like `message["choices"][0]['message']` is correct, did you see the delta key used by some runtime?

Comment (dagrayvid) [reply plugins/openai_plugin.py:108]: Nevermind, I understand now! Delta is used for streaming chunks, message is used for non-streaming. Thanks for the fix!
https://platform.openai.com/docs/api-reference/chat/streaming

File: README.md (hunk 1 above)
Comment (ccamacho) [inline README.md:42]: the output file is configured in the config file under the 'output' key, I wouldn't hardwire this, but explain where the file comes from.

File: README.md [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
+```
+python3 -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
+```
+
 **Running the Tool**:
+
+- Change/Update the configuration file `config.yaml` as needed for your test. Look at the appropriate plugin for the configuration options.
+- Run the tool with the following command: `python load_test.py -c config.yaml`
```
Comment (ccamacho) [inline README.md:22]: python3 maybe?  I suppose this depends on the interpreter

File: plugins/openai_plugin.py (hunk 2 above)
Comment (ccamacho) [reply plugins/openai_plugin.py:108]: Good catch! I tested streaming before.

File: performance_visualization.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import json
```
Comment (ccamacho) [inline performance_visualization.py:1]: No sure of adding this as the scope of llm-load-test, can you push this in another PR for for further discussion?

**Comment by ccamacho** [comment]
@HunterGerlach this is amazing! Thanks a lot for your contributions in llm-load-test!!!!

File: performance_visualization.py (hunk 4 above)
Comment (dagrayvid) [reply performance_visualization.py:1]: It's already used in utils.py, and it's a common package

Comment (ccamacho) [reply performance_visualization.py:1]: nana I mean the visualization script, so we can merge this before and talk about the visualization later

Comment (dagrayvid) [reply performance_visualization.py:1]: Ah! Sorry for misunderstanding, I agree. @HunterGerlach can we merge this PR without the visualization code, and have a separate PR for that?

Comment (HunterGerlach) [reply performance_visualization.py:1]: Absolutely. I actually hesitated including it initially, so that's fine.

**Comment by ccamacho** [comment]
I just included a commit to remove the extra visualization, lets find the correct place for that script.

**Pull request merged**